### ISSUE TITLE
Intermittent connection failures from Mac

### ISSUE THREAD

**Issue by sundeeptessell** [tailscale/tailscale#4305]
### What is the issue?

Every 2-3 hours, when I am connected to tailscale, I see the connections terminate and I get reconnected after 5-10 min. Somtimes it even takes 30 min to connect back. The connections resume if I disconnect the tailscale, so surely the issue is happening due to tailscale. 

Please let me know if any logs need to be attached.

### Steps to reproduce

No clear steps to reproduce. I connect to the tailscale and after couple of hrs, randomly, it disconnects me. 

### Are there any recent changes that introduced the issue?

No.

### OS

macOS

### OS version

12.3

### Tailscale version

1.22.1

### Bug report

BUG-6748e4a7f0e74b0db7045323259adec85713974641ad8e3a693c0c0eb62aaf2c-20220328140409Z-0d8b477bea216aea

**Comment by DentonGentry** [comment]
Slightly before the bugreport:
```
2022-03-28 19:23:16.149535 +0530 +0530: magicsock: [0xc000440000] derp.Recv(derp-6): derphttp.Client.Recv connect to region 6 (blr): context deadline ex
ceeded: read tcp4 127.0.0.1:58983->127.0.0.1:10011: use of closed network connection
2022-03-28 19:23:16.149743 +0530 +0530: derp-6: backoff: 5724 msec

2022-03-28 19:23:17.698192 +0530 +0530: dns: resolver: forward: DoH error from 8.8.8.8: Post "https://dns.google/dns-query": EOF
2022-03-28 19:23:17.698374 +0530 +0530: dns: resolver: forward: DoH error from 8.8.8.8: Post "https://dns.google/dns-query": EOF
```

but I don't see a reason for it.

**Comment by DentonGentry** [comment]
In 1.28 we switched the macOS client to install a single 100.64.0.0/10 route, instead of individual /32 routes for each Tailscale destination. This avoided ERR_NETWORK_CHANGED every time another node on the tailnet would come or go, which I think is the root cause of this issue.

Marking as fixed in 1.28, please reopen if it still happens, or open a new issue if the symptoms are different.

https://github.com/tailscale/tailscale/releases/tag/v1.28.0
